trunk/NDHMS/Routing/Reservoirs/Persistence_Level_Pool_Hybrid/module_persistence_levelpool_hybrid.F

```diff
@@ -0,0 +1,350 @@
+
+        end if
+
+    end subroutine hybrid_struct_init
+
+
+    !Level Pool Destructor
+    subroutine hybrid_struct_destroy(this)
+        implicit none
+        class(persistence_levelpool_hybrid_struct), intent(inout) :: this ! object being destroyed
+    end subroutine hybrid_struct_destroy
+
+
+    ! Subroutine for running release for a hybrid reservoir
+    subroutine run_hybrid_release(reservoir_data, previous_timestep_inflow, inflow, &
+        lateral_inflow, water_elevation, outflow, routing_period)
+        implicit none
+        class(persistence_levelpool_hybrid_struct), intent(inout) :: reservoir_data
+        real, intent(in)    :: previous_timestep_inflow ! cubic meters per second (cms)
+        real, intent(in)    :: inflow                   ! cubic meters per second (cms)
+        real, intent(in)    :: lateral_inflow           ! cubic meters per second (cms)
+        real, intent(inout) :: water_elevation          ! meters
+        real, intent(out)   :: outflow                  ! cubic meters per second (cms)
+        real, intent(in)    :: routing_period           ! seconds
+        real                :: delta_storage            ! timestep change in storage (cubic meters)
+        real                :: local_water_elevation    ! water elevation passed to levelpool (meters AMSL)
+        real                :: levelpool_outflow        ! cubic meters per second (cms)
+        logical             :: max_storage_reached      ! flag for when max storage is reached
+        integer             :: gage_lookback_seconds    ! seconds between current model
+                                                        ! time and the time a gage
+                                                        ! discharge is read
+
+        max_storage_reached = .false.
+        gage_lookback_seconds = 0
+
+        ! Update input variables
+        reservoir_data%input%inflow = inflow
+        reservoir_data%input%lateral_inflow = lateral_inflow
+
+        ! The initialization and management of water elevation might diverge between the global state of the model
+        ! and the internal state. If water elevation is refactored to no longer be a global state, then this check
+        ! is not necessary. This currently allows restarts to work without the reservoir managing its own restart.
+        if (reservoir_data%state%water_elevation .ne. water_elevation) then
+
+            ! Update state variables
+            reservoir_data%state%water_elevation = water_elevation
+
+            reservoir_data%state%current_storage = (reservoir_data%state%water_elevation - reservoir_data%parameters%orifice_elevation) * &
+            reservoir_data%parameters%lake_area
+
+        end if
+
+        local_water_elevation = reservoir_data%state%water_elevation
+
+        ! If update time to read new timeslice gage discharges. This condition is only met in retrospective runs.
```

Comment:
Is it not met in the analysis run?

Reply:
Our terminology is not quite synchronized...I think this comment really should say "Happens exactly once in forecast modes (at the first time step/call) and recurs every update_interval in non forecast modes.  These are determined by the name-list parameters configured with the various simulation types." with perhaps a quick example of the name-list configuration for a forecast and non-forecast run as an example...

Reply:
@jdmattern-noaa fixing description